### PR TITLE
add option to only process files that match a filter

### LINKED ISSUE #10: How do I configure this plugin to process only my amp-html pages?
Hi

My project contains a mix of html pages and amp-html pages. How do I configure this plugin to process only my amp-html pages?

Ideally I'd want to process only files/directories matching a regular expression e.g. `*-amp.html` or `_site/*/amp/*`.

Thanks!

### PR DESCRIPTION
**Pull request by patrickkettner**
fixes #10

@sebastianbenz I opted to merge the options object to avoid doing an existence check. Feels weird to have duplicate code in two places, so if you have an opinion on how it should be modularized or replaced, let me know.

### CODE REVIEW

File: src/transforms/ampValidation.js [hunk 1]
```diff
@@ -54,4 +68,4 @@ function createErrorMessage(error, outputPath) {
   return msg;
 }
 
-module.exports = ampValidationTransform;
+module.exports = ampValidationTransforom;
```
Comment (sebastianbenz) [inline src/transforms/ampValidation.js:71]: typo? `ampValidationTransforom`

Comment (patrickkettner) [reply src/transforms/ampValidation.js:71]: all mistakes in my life are blamed on the butterfly keyboard.

**Comment by patrickkettner** [comment]
> wdyt

Im good with whatever, but Im not 100% sure I understand. In your suggestion,  would `optimizationTransform ` be [this](https://github.com/ampproject/eleventy-plugin-amp/blob/7766995bd21530b206b6104cf0149774c85e6a3b/src/transforms/ampTransform.js#L23-L31)  and be what is exported from `src/transforms/ampTransform.js` ?

Another option would be to move the option handling logic could be plunked [here](https://github.com/ampproject/eleventy-plugin-amp/blob/7766995bd21530b206b6104cf0149774c85e6a3b/src/.eleventy.js#L21)

**Comment by sebastianbenz** [comment]
I was thinking:

```
const defaultOptions = {
  filter: /.*/
}

const ampTransformWith = (transform, eleventyConfig, providedOptions = {}) => {

  const options = Object.assign(defaultOptions, providedOptions)
  let filter

  try {
    filter = new RegExp(options.filter)
  } catch (e) {
    throw new Error(`filter needs to be a valid RegExp, provided : ${options.filter}`)
  }
  eleventyConfig.addTransform('amp', async (content, outputPath) => {
    if (!outputPath.endsWith('.html') || !filter.test(outputPath)) {
      return content;
    }

    const transformedContent = await transform(content, {
      outputPath,
    });

    return transformedContent;
    });
};

module.exports = ampTransformWith
```

> Another option would be to move the option handling logic could be plunked here

This would good as well.

**Comment by patrickkettner** [comment]
@sebastianbenz sorry to question the paint on the ol bicycle shed, but do you expect people to ever use the transforms individually (e.g. `require("@ampproject/eleventy-plugin-amp/src/transforms/ampValidation.js")`? If so, we would want to have something like a helper func that does the option handling that is called from the existing transforms, rather than wrapping these, right?

**Comment by patrickkettner** [comment]
updated the PR to reflect what I was thinking.

File: src/transforms/ampTransform.js [hunk 2]
```diff
@@ -16,11 +16,15 @@
...
+
+const ampTransform = (eleventyConfig, providedOptions = {}) => {
+
+  const options = processOptions(providedOptions);
 
-const ampTransform = (eleventyConfig, options = {}) => {
   const ampOptimizer = createAmpOptimizer(options);
   eleventyConfig.addTransform('amp', async (content, outputPath) => {
-    if (!outputPath.endsWith('.html')) {
+    if (!outputPath.endsWith('.html') || !options.filter.test(outputPath)) {
```
Comment (sebastianbenz) [inline src/transforms/ampTransform.js:27]: We could theoretically move the whole expression into a function on the options object:

```
if (options.isAmp(path)) {
   ...
}
```

Then we could remove all duplication between the two transforms and give users even more control over which files are processed.

**Pull request merged**